app/controllers/user_credentials_controller.rb

```diff
@@ -14,6 +14,22 @@ def index
     end
   end
 
+  def import
+    file = params[:file]
+    content = JSON.parse(file.read)
+    new_credentials = content.map do |hash|
+      current_user.user_credentials.build(hash.slice("credential_name", "credential_value", "mode"))
+    end
+
+    respond_to do |format|
+      if new_credentials.map(&:save).all?
+        format.html { redirect_to user_credentials_path, notice: "The file was successfully uploaded."}
+      else
+        format.html { redirect_to user_credentials_path, notice: 'One or more of the uploaded was not imported due to duplicate credential name error'}
```

Comment:
Maybe?

'One or more of the uploaded credentials was not imported due to an error.  Perhaps an existing credential had the same name?'